packages/contracts/test/FeaturedVotingContract.t.sol

```diff
@@ -0,0 +1,459 @@
+        vm.expectRevert(bytes('no ongoing vote'));
+        featuredVotingContract.finalizeVoting();
+    }
+
+    function test_RevertWhen_VoteStillOngoing() public {
+        vm.prank(votingContract);
+        directoryContract.addCommunity(communityID1);
+
+        featuredVotingContract.initializeVoting(communityID1, 100);
+        vm.expectRevert(bytes('vote still ongoing'));
+        featuredVotingContract.finalizeVoting();
+    }
+
+    function test_FinalizeVoting() public {
+        vm.prank(votingContract);
+        directoryContract.addCommunity(communityID1);
+
+        featuredVotingContract.initializeVoting(communityID1, 100);
+        skip(votingWithVerificationLength + 1);
+        vm.expectEmit(false, false, false, false);
+        emit VotingFinalized();
+        featuredVotingContract.finalizeVoting();
+
+        FeaturedVotingContract.Voting[] memory votings = featuredVotingContract.getVotings();
+        assertEq(votings.length, 1);
+        assertEq(votings[0].id, 1);
+        assert(votings[0].finalized);
+    }
+}
+
+contract CastVotes_Test is FeaturedVotingContract_Test {
+    function setUp() public virtual override {
+        FeaturedVotingContract_Test.setUp();
+    }
+
+    function test_RevertWhen_NoOngoingVote() public {
+        FeaturedVotingContract.SignedVote[] memory votes = new FeaturedVotingContract.SignedVote[](0);
+        vm.expectRevert(bytes('no ongoing vote'));
+        featuredVotingContract.castVotes(votes);
+    }
+
+    function test_RevertWhen_VotingHasBeenClosedAlready() public {
+        vm.prank(votingContract);
+        directoryContract.addCommunity(communityID1);
+
+        featuredVotingContract.initializeVoting(communityID1, 100);
+        skip(votingWithVerificationLength + 1);
+
+        FeaturedVotingContract.SignedVote[] memory votes = new FeaturedVotingContract.SignedVote[](1);
+        votes[0] = _createSignedVote(bobsKey, bob, communityID1, 200, block.timestamp);
+        vm.expectRevert(bytes('vote closed'));
+        featuredVotingContract.castVotes(votes);
+    }
+
+    function test_CastVotes_EmitInvalidSignature() public {
+        vm.prank(votingContract);
+        directoryContract.addCommunity(communityID1);
+        featuredVotingContract.initializeVoting(communityID1, 100);
+
+        FeaturedVotingContract.SignedVote[] memory votes = new FeaturedVotingContract.SignedVote[](1);
+        // create broken signature with bob's key but alice's address
+        votes[0] = _createSignedVote(bobsKey, alice, communityID1, 200, block.timestamp);
+
+        vm.expectEmit(false, false, false, true);
+        emit InvalidSignature(communityID1, alice);
+        featuredVotingContract.castVotes(votes);
+    }
+
+    // // it seems impossible to come up with a scenario in which `castVotes()`
+    // // would emit `CommunityFeaturedRecently`, so this test is disabled
+    // function test_CastVotes_CooldownPeriodHasNotPassed() public {
```

Comment:
@osmaczko would be wrong but it look like we can remove some code in `castVotes()`.
I couldn't get a scenario to work in which `CommunityFeaturedRecently` would be emitted via `castVotes`.

Reply:
https://github.com/status-im/community-dapp/pull/60

Reply:
Cool, will review and update this PR accordingly if it's the right path forward

Reply:
Tested your PR and ran the enabled foundry test on top of it as well.
Everything's green!